packages/app/src/app/pages/NewDashboard/Components/Selection/ContextMenu.tsx

```diff
@@ -41,16 +47,39 @@ export const ContextMenu: React.FC<IContextMenuProps> = ({
   selectedIds,
   sandboxes,
   folders,
+  repos,
   setRenaming,
   createNewFolder,
   createNewSandbox,
+  page,
 }) => {
   if (!visible) return null;
 
   const selectedItems: Array<
-    DashboardFolder | DashboardSandbox | DashboardTemplate
+    | DashboardFolder
+    | DashboardSandbox
+    | DashboardTemplate
+    | DashboardRepo
+    | DashboardNewMasterBranch
   > = selectedIds.map(id => {
     if (id.startsWith('/')) {
+      if (repos.length) {
```

Comment:
Should we maybe change `repos?: ` to `repos: ` in the props? Since we're assuming here that `repos` cannot be `undefined`.

Reply:
my bad, also checked if repos exist now